cai_causal_graph/graph_components.py

```diff
@@ -193,12 +193,22 @@ def to_dict(self, include_meta: bool = True) -> dict:
         node_dict = {
             'identifier': self.identifier,
             'variable_type': self.variable_type,
+            'node_class': self.__class__.__name__,
         }
         if include_meta:
             node_dict['meta'] = deepcopy(self.meta)   # type: ignore
 
         return node_dict
 
+    @classmethod
+    def from_dict(cls, node_dict: dict) -> Node:
+        """Return a `cai_causal_graph.graph_components.Node` instance from a dictionary."""
```

Comment:
Should we assert `'node_class'` is correct?

Reply:
Looking at unit tests you do allow it and I think it is fine as the dicts are compatible. Maybe we just warn if they try to make one saved from other class?